scripts/process_uploads.py

```diff
@@ -35,28 +37,26 @@ def determine_samples(proj):
 
     # Determines which sequences have had their metadata fields updated
     # (This metadata is updated on initial notification of upload)
-    for sample in samples_without_analysis['sample_ids']:
-        try:
-            seq_id = int(seqapi.get_sequence_id_from_sample_id(sample, proj))
-            seq_entry = seqapi.get_sequence_by_id(seq_id, proj)
-            full_external_id_batch_mapping = []
-
-            # This will return a dictionary. Check if a dictionary has a key.
-            if seq_entry['meta'] is not None:
-                if 'gvcf' in seq_entry['meta']:
-                    batch = seq_entry['meta']['gvcf'][-1]['batch']
-                    samples_with_sequencing_meta.append(sample)
-                    external_map = sapi.get_sample_id_map_by_internal(proj, [sample])
-                    full_external_id_batch_mapping.append(
-                        {
-                            'internal_id': sample,
-                            'external_id': external_map[sample],
-                            'batch': batch,
-                        }
-                    )
-
-        except ServiceException:
-            print(f'Sequencing for {sample} not found')
+    for internal_sample_id in samples_without_analysis['sample_ids']:
+        seq_id = int(seqapi.get_sequence_id_from_sample_id(internal_sample_id, proj))
+        seq_entry = seqapi.get_sequence_by_id(seq_id, proj)
```

Comment:
I'm not really a fan of lots of web requests in the loop, I've just added `get_sequences_by_internal_sample_ids` (will require regeneration of SM) unfortunately to hopefully dramatically reduce the amount of web requests.

```suggestion
    sequences = seqapi.get_sequences_by_internal_sample_ids(
        project=proj, request_body=samples_without_analysis['sample_ids']
    )
    for seq_entry in sequences:
        internal_sample_id = seq_entry.sample_id
```